controller/aks-cluster-config-handler.go

```diff
@@ -604,12 +605,28 @@ func BuildUpstreamClusterState(ctx context.Context, secretsCache wranglerv1.Secr
 		}
 	}
 
+	// set addon monitoring profile
+	addonProfile := clusterState.AddonProfiles
+	if addonProfile != nil {
+		upstreamSpec.Monitoring = addonProfile["omsagent"].Enabled
+		logAnalyticsWorkspaceResourceID := addonProfile["omsagent"].Config["logAnalyticsWorkspaceResourceID"]
+
+		matchWorkspaceGroup := regexp.MustCompile("/resourcegroups/(.+?)/")
+		logAnalyticsWorkspaceGroup := matchWorkspaceGroup.FindStringSubmatch(to.String(logAnalyticsWorkspaceResourceID))[1]
+		upstreamSpec.LogAnalyticsWorkspaceGroup = to.StringPtr(logAnalyticsWorkspaceGroup)
+
+		matchWorkspaceName := regexp.MustCompile("/workspaces/(.+?)")
+		logAnalyticsWorkspaceName := matchWorkspaceName.FindStringSubmatch(to.String(logAnalyticsWorkspaceResourceID))[1]
```

Comment:
Same here?

Reply:
It will be importing this data from Azure API, we can assume that it was already validated